pylint/checkers/base/basic_checker.py

```diff
@@ -366,20 +366,19 @@ def _check_using_constant_test(
                     call_inferred = inferred.infer_call_result()
                 elif isinstance(inferred, nodes.Lambda):
                     call_inferred = inferred.infer_call_result(node)
+                # Just forcing the generator to infer all elements.
+                # astroid.exceptions.InferenceError are false positives
+                # see https://github.com/PyCQA/pylint/pull/8185
+                if call_inferred is not None:
+                    call_inferred = list(call_inferred)
```

Comment:
```suggestion
                    call_inferred = list(inferred.infer_call_result())
                elif isinstance(inferred, nodes.Lambda):
                    call_inferred = list(inferred.infer_call_result(node))
```

I can't suggest to place the comment above here within the Github UI, but this should be functionally the same right?